PR 65

ericsorenson opened the pull request:
Stories & Bugs
- [8009 BUG: Edit Docket Entry: Service Date datepicker not clearing when documentType is updated](https://github.com/flexion/ef-cms/issues/8009) 
- [7954 BUG: Resetting trial status throws error](https://github.com/flexion/ef-cms/issues/7954) 
- [7887 Public: Sortable Today's Orders (Smaller story)](https://github.com/flexion/ef-cms/issues/7887) 
- [7863 Docket Clerk: Ability to add additional information when adding a Minute Sheet to the docket record](https://github.com/flexion/ef-cms/issues/7863) 
- [7855 Spike: Find all the instances where the app is failing quietly](https://github.com/flexion/ef-cms/issues/7855) 
- [5311 IRS: Summary of all service items](https://github.com/flexion/ef-cms/issues/5311) 

Refactoring Tasks
- https://trello.com/c/3gH4hRRi

Migrations
- `0024-docket-entry-docket-number-served-parties-code`

Deployment Notes
- Upgrade to Terraform 14.8

kkoskelin committed:
devex 851: wip [skip ci]

Commit:
922: updated adding coversheet to document

Commit:
922: updated archive draft document

Commit:
922: updated complete docket entry qc

ericsorenson committed:
Merge remote-tracking branch 'upstream/staging' into from-ustaxcourt

ericsorenson committed:
package upgrades

Commit:
922: updated get download policy url

Commit:
devex 992: jsdoc

Commit:
922: updated file court issued docket entry

Commit:
922: updated file court issued order

Commit:
922: updated file docket entry

Commit:
922: updated file external document

Commit:
Merge branch 'develop' of github.com:flexion/ef-cms into devex-922-order

Commit:
922: updated file external document for consolidated cases

Commit:
922: updated generate printable filing receipt

Commit:
922: updated get document contents for docket entry

Commit:
922: updated get document download

Commit:
922: updated get upload policy

Commit:
922: updated opinion advanced search

Commit:
922: updated order advanced search

Commit:
922: fix tests

Commit:
922: fix addCoversheetInteractor usages with updated appContext param

rachelschneiderman committed:
Devex 892: try adding pendingColorSwitch value to deploy table

kkoskelin committed:
Devex 851: hoist updates to practitioners on cases to the helper

rachelschneiderman committed:
Devex 892: fixing env variable access

TomElliottFlexion committed:
7510: added clearServiceDateAction to clear serviceDate datepicker when documentType is updated

rachelschneiderman committed:
Add AWS credentials to pending color switch step

TomElliottFlexion committed:
8009: Moved form clearing logic to datepicker component

Commit:
922: fixed unit / integration tests

Commit:
922: fixed versioned lambdas

rachelschneiderman committed:
Change how we access the ENV for pending-color-switch script

rachelschneiderman committed:
Merge pull request #918 from flexion/pr-64

Latest to migration

Commit:
922: fixed integration test

klohman committed:
Update release-notes.md

rachelschneiderman committed:
Devex 892: Add back tests to circle config

ericsorenson committed:
upgrade terraform to 0.14.8

klohman committed:
Update release-notes.md

TomElliottFlexion committed:
8009: Added cypress spec to test updated datepicker logic

TomElliottFlexion committed:
8009: Removed comments

klohman committed:
Merge pull request #8011 from flexion/pr-64-release-notes

Update release-notes.md

rachelschneiderman committed:
Merge pull request #8012 from flexion/892-pending-color-switch

Devex 892: Add `pending_color_switch` flag to deploy table

kkoskelin committed:
5311: endpoint, lambda, interactor WIP [skip ci]

ericsorenson committed:
Merge branch 'develop' into from-ustaxcourt

ericsorenson committed:
Update package.json

TomElliottFlexion committed:
Merge pull request #8007 from flexion/7510-serviceDate-feedback

8009: service date feedback

kkoskelin committed:
Merge pull request #8006 from flexion/devex-851-case-helper-1.8

Devex 851 case helper 1.8

Commit:
fix interactor/proxy usage

sutt0n committed:
Merge pull request #7996 from flexion/devex-922-docs

Devex 922: separate app context param in lambdas - documents pt 1

Commit:
Merge branch 'develop' of github.com:flexion/ef-cms into devex-922-docs-2

Commit:
922: remove unnecessary object spread operator

kkoskelin committed:
5311: interactor and tests

ericsorenson committed:
temporarily remove locks

ericsorenson committed:
Merge pull request #7991 from flexion/from-ustaxcourt

From ustaxcourt/staging

Commit:
922: updated remove pdf from docket entry

Commit:
922: updated remove signature from document

Commit:
922: updated save signed document

Commit:
922: updated save externally filed document

Commit:
922: updated strike docket entry

Commit:
922: updated update court issued docket entry

Commit:
922: updated update court issued order

Commit:
922: updated update docket entry meta

Commit:
922: updated update docket entry on case

Commit:
922: updated validate pdf

Commit:
Merge branch 'develop' of github.com:flexion/ef-cms into devex-922-docs-3

rosiehsmith committed:
5311: WIP writing elastic search query for reconciliation report [skip ci]

codyseibert committed:
writing unit tests over the date picker component

codyseibert committed:
removing value from DatePickerComponent since it is not used

sutt0n committed:
Merge pull request #8013 from flexion/devex-922-docs-2

Devex 922: separate app context param in lambdas - documents pt 2

Commit:
922: added missing jsdoc description

Commit:
strip double quotes

Commit:
922: address comment

rosiehsmith committed:
5311: [skip ci] working query in getReconciliationReport persistence, tbd reindexing needed if adding servedPartiesCode

Commit:
922: addressed comment

sutt0n committed:
Merge branch 'develop' into devex-922-docs-3

Commit:
added .terraform.lock.hcl

sutt0n committed:
Merge pull request #8014 from flexion/devex-922-docs-3

Devex 922: separate app context param in lambdas - documents pt 3

rosiehsmith committed:
5311: [skip ci] WIP adding integration test to get report endpoint

Commit:
fixed typo, tf version

Commit:
Merge branch 'develop' of github.com:flexion/ef-cms into tf-14

kkoskelin committed:
5311: record updated docket entry, verify in report

codyseibert committed:
5311 - adding tests

rosiehsmith committed:
5311: extracted assignCaseCaptionFromPersistence and updated test

codyseibert committed:
Merge branch 'develop' into devex-date-picker-unit-tests

TomElliottFlexion committed:
7954: Updated TrialSessionWorkingCopy validation to allow trialStatus to be reset

Commit:
added missing log groups

codyseibert committed:
lock uswds due to enzyme error

kkoskelin committed:
5311: fixing linting, removing comment

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into 5311-service-item-summary

# Conflicts:
#	package-lock.json
#	shared/src/business/useCases/addCoversheetInteractor.js

TomElliottFlexion committed:
Merge pull request #8018 from flexion/7954-BUG-validation

7954: Updated TrialSessionWorkingCopy validation to allow trialStatus…

codyseibert committed:
revert uswds lockdown; using jsdom for testEnvironment so that enzyme works correctly

codyseibert committed:
addressing pr comments

rosiehsmith committed:
Merge pull request #8017 from flexion/5311-service-item-summary

5311 service item summary

codyseibert committed:
Merge pull request #8015 from flexion/devex-date-picker-unit-tests

DevEx - date picker unit tests

mmarcotte committed:
Merge pull request #919 from flexion/latest-to-migration

Latest to migration

rosiehsmith committed:
7863: added description for type f

rosiehsmith committed:
7863: add showOptionalFreeText to helper test

rosiehsmith committed:
Merge pull request #8022 from flexion/7863-description-typef

7863: added description for type f

rosiehsmith committed:
7863-dod: npm update

rosiehsmith committed:
Merge pull request #8023 from flexion/7863-dod

7863-dod: npm update

kkoskelin committed:
Devex 851: use helper method for individual persistence calls of different kinds of objects

ericsorenson committed:
Merge remote-tracking branch 'upstream/staging' into from-ustaxcourt

ericsorenson committed:
upgrade terraform from 0.13.6 to 0.14.8

ericsorenson committed:
Merge remote-tracking branch 'origin/tf-14' into upgrade-terraform-to-0.14.8

kkoskelin committed:
Restrict USWDS to patch-level versions of ~2.10

Upgrading to USWDS version ^2.11.x introduces unexpected changes to icons, so we're keeping it as ~2.10 for now.
Symptom of upgrading to 2.11 seen below:

rosiehsmith committed:
7863: update to "Description" instead of "Enter description"

rosiehsmith committed:
7863: add class to docket-preview in EditDocketEntryMeta to have visible overflow.

kkoskelin committed:
Merge pull request #8030 from flexion/uswds-patchlevel-lock

Restrict USWDS to patch-level versions of ~2.10

rosiehsmith committed:
Merge pull request #8032 from flexion/7863-ux-feedback

7863 ux feedback

ericsorenson committed:
Merge pull request #8027 from flexion/from-ustaxcourt

From ustaxcourt/staging

TomElliottFlexion committed:
7887: Added sort selector to today's orders, updated associated interactor, sequence, proxy, and infrastructure. Began writing tests. WIP

kkoskelin committed:
7887: options to change sort order of today's orders

kkoskelin committed:
7887: tests, actions, sequences for altering sort order and column WIP [skip ci]

dsalls committed:
7887: set orderType to ascending for filingDate or numberOfPages [skip-ci]

ericsorenson committed:
Merge pull request #932 from flexion/pr-65

Latest to Migration

kkoskelin committed:
Merge pull request #8024 from flexion/devex-851-case-helper-2.0

Devex 851: use helper method to create case

ericsorenson committed:
Merge pull request #8031 from flexion/upgrade-terraform-to-0.14.8

Upgrade terraform to 0.14.8

ericsorenson committed:
Merge remote-tracking branch 'upstream/staging' into from-ustaxcourt

dsalls committed:
7887: reset todays orders page and refactoring. Temporary removal of sort by number of pages until ES mapping updated [skip-ci]

ericsorenson committed:
Merge pull request #8034 from flexion/from-ustaxcourt

From ustaxcourt/staging

klohman commented:
## PR 65 Release Notes

### Internal Only 

**Bug Fix** The Service Date field now clears when updating a docket entry to Certificate of Service or Amended Certificate of Service. 

**Bug Fix** On the Judge’s Working Copy, you can now reset the trial status to unselected without an error. 

**New Feature** Docket clerks can now enter optional additional text as part of the docket entry for all Minute Sheets. 

### External / Public 

**New Feature** A summary of all documents served on Respondent is now available to the IRS. 

**New Feature** Today’s Orders can now be sorted by time served and page count.

klohman committed:
Update release-notes.md

ericsorenson committed:
Merge remote-tracking branch 'upstream/migration' into pr-65

kkoskelin committed:
7887: updated tests for changing sort order

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into 7887-sort-todays-orders

# Conflicts:
#	package-lock.json

kkoskelin committed:
7887:  adding integer indexing for "numberOfPages"

adding sorting options for pages

mmarcotte committed:
Merge pull request #936 from flexion/pr-65

Latest to ustaxcourt/migration

kkoskelin committed:
7887: updating tests to match new usage

kkoskelin committed:
7887: undoing change to interactor call

confused changes to interactor with calls to the proxy

kkoskelin committed:
7887: updating sort option count

we enabled all four options instead of just two

kkoskelin committed:
7887: cleanup of constants, separation of concerns

dsalls committed:
7887: update failing test

klohman committed:
Merge pull request #8041 from flexion/pr-65-release-notes

Update release-notes.md

kkoskelin committed:
Merge pull request #8042 from flexion/7887-sort-todays-orders

7887: sort todays orders

mmarcotte committed:
Merge pull request #942 from ustaxcourt/staging

Merge Staging into Migration

ericsorenson committed:
Merge branch 'pr-65' into from-ustaxcourt

ericsorenson committed:
Merge remote-tracking branch 'upstream/staging' into from-ustaxcourt

ericsorenson committed:
package upgrades

ericsorenson committed:
Merge pull request #8048 from flexion/from-ustaxcourt

From ustaxcourt/staging

ericsorenson committed:
Merge branch 'develop' into pr-65

ericsorenson committed:
Merge branch 'staging' into pr-65

ericsorenson committed:
Merge branch 'migration' into pr-65

ericsorenson committed:
Merge pull request #945 from flexion/pr-65

Latest to Migration

mmarcotte committed:
Merge pull request #947 from ustaxcourt/staging

Merge Staging into Migration

mmarcotte committed:
Merge pull request #949 from ustaxcourt/staging

Introduce scheduled test into migration branch

ericsorenson committed:
Merge remote-tracking branch 'upstream/staging' into pr-65

ericsorenson committed:
upgrade deps

ericsorenson committed:
Merge branch 'develop' into pr-65

kkoskelin committed:
5311: remove match predicates on pk/sk

elasticsearch indexes should no longer contain "match" queries for pk or sk because column types have changed.

kkoskelin committed:
Merge pull request #8052 from flexion/5311-reconciliation-mapping-query

5311: remove match predicates on pk/sk

ericsorenson committed:
Merge remote-tracking branch 'upstream/migration' into develop

kkoskelin committed:
5311: prevent undefined users from causing errors

return not-authorized value instead.

kkoskelin committed:
5311 fixing authorization bug scenario WIP

[skip ci]

kkoskelin committed:
5311: backup method to get docketNumber [skip ci]

extract it from PK if not on docketEntry itself.

matthopson committed:
5311 - Ensure docketNumber gets added to signed stip docket entry

matthopson committed:
5311 - DocketEntry Entity - set servedPartiesCode when setting doc as served

matthopson committed:
5311 - Migration - update docketNumber and servedPartiesCode on docketEntry items

matthopson committed:
5311 - Case Entity - Ensure docketNumber gets added to docketEntry items via addDocketEntry method

matthopson committed:
Merge pull request #8060 from flexion/5311-es-search-fix

5311 - Reconciliation Report Fixes

kkoskelin committed:
5311: updating v2 API documentation

added /v2/reconciliation-report information

kkoskelin committed:
Merge pull request #8061 from flexion/5311-es-search-fix

5311: updating v2 API documentation

mmarcotte commented on docs/api/v2.yaml:
```diff
@@ -83,6 +83,37 @@ paths:
+            type: string
+      responses:
+        '200':
+          description: 
```
Just flagging so it doesn't get lost, but I this needs a string value to pass validation. I'm pretty sure @kkoskelin has a fix incoming.

mmarcotte commented on docs/api/v2.yaml:
```diff
@@ -83,6 +83,37 @@ paths:
+  /reconciliation-report/{date}:
+    get:
+      summary: fetches a list of docket entries served on a date
+      operationId: 
```
Just flagging so it doesn't get lost, but I this needs a string value to pass validation. I'm pretty sure @kkoskelin has a fix incoming.

kkoskelin replied:
in the queue: https://github.com/flexion/ef-cms/pull/8067

kkoskelin committed:
5311: yaml update and validation

added non-empty operationId and description

matthopson committed:
Upgrade sonar-scanner-cli to 4.6.0

mmarcotte commented on shared/src/business/useCases/getReconciliationReportInteractor.js:
```diff
@@ -0,0 +1,105 @@
+    e.docketNumber = docketNumber;
+    return e.docketNumber;
+  });
+  const casesDetails = await applicationContext
+    .getPersistenceGateway()
+    .getCasesByDocketNumbers({ applicationContext, docketNumbers });
+
+  docketEntries.forEach(docketEntry => {
+    docketEntry.caseCaption = casesDetails.find(
+      detail => detail.docketNumber === docketEntry.docketNumber,
+    ).caseCaption;
+  });
```
Added [refactoring opportunity for performance](https://trello.com/c/vRyVUZag/934-performance-optimization-in-reconciliationreport).

matthopson committed:
Enable verbose logging for sonar

mmarcotte commented on shared/src/persistence/elasticsearch/getReconciliationReport.js:
```diff
@@ -0,0 +1,69 @@
+          'servedPartiesCode',
+        ],
+        query,
+        size: 5000,
```
Something to keep an eye on. What happens if we serve more than 5000 documents? Based on my observations in Mig, this number is around 1,000. 

Nothing that requires changing because I think this should suffice, but we may need to consider what happens if the total results is > 5000. Yes, we can increase to 10k or use ES's scroll feature. And we may need to consider pagination in the report itself.

matthopson committed:
Disable sonar cloud for the time being

matthopson committed:
Merge pull request #8067 from flexion/5311-swagger-update

5311: yaml update and validation

matthopson committed:
Merge pull request #8070 from flexion/upgrade-sonar-4-6-0

Disable SonarQube
